content/post/111-js-ipfs-0-51-0.md

```diff
@@ -0,0 +1,220 @@
+
+With `js-IPFS@0.51.0` types are now a first-class citizen within the codebase.  The project has not gone so far as to convert to TypeScript - we do not wish knowledge of TypeScript to become a prerequisite to contributing to `js-IPFS` development, instead all types are defined using [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) comments and are built and validated during development and Continuous Integration.
+
+<p style="max-width:1000px;margin-left:auto;margin-right:auto;">
+<img src="/111-js-ipfs-0-51-0/types.png">
+</p>
+
+## SECIO Removal
+
+The [deprecated](https://blog.ipfs.io/2020-08-07-deprecating-secio/) security transport [SECIO](https://github.com/libp2p/specs/blob/master/secio/README.md) has finally been removed in `js-IPFS@0.51.0`.  It's successor, [Noise](https://github.com/libp2p/specs/blob/master/noise/README.md) was first released with `js-IPFS@0.47.0` and is now the default security transport used by libp2p.
+
+`go-IPFS@0.7.0` recently also removed SECIO support after introducing Noise in `go-IPFS@0.6.0`, which means `js-IPFS@0.51.0` and `go-IPFS@0.7.0` can only talk to other nodes running `go-IPFS@0.6.0` or `js-IPFS@0.47.0` or later.
+
+## 📌 Custom DAG Formats
+
+At it's heart IPFS is about files, which means [UnixFS](https://github.com/ipfs/specs/blob/master/UNIXFS.md) and in terms of [IPLD Formats](https://github.com/ipld/interface-ipld-format) means [dag-pb](https://github.com/ipld/js-ipld-dag-pb) and [ipld-raw](https://github.com/ipld/js-ipld-raw).
+
+IPFS gives you access to all the underlying components though, allowing you to use [dag-cbor](https://github.com/ipld/js-ipld-dag-cbor) to connect your data structures together and have them accessible and distributable across the internet through [libp2p](https://github.com/libp2p/js-libp2p).
+
+But what if these codecs don't meet your requirements, what if you want to develop a new codec or use less common ones like [dag-jose](https://github.com/ceramicnetwork/js-dag-jose)?
+
+We don't bundle every available codec with the default `js-IPFS` install because some of them pull in large dependency trees which will unnecessarily slow down the install for people who aren't going to use them.
+
+Previously you've been able to use [custom IPLD formats](https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client#ipld-formats) with `ipfs-http-client` and when running `ipfs` as an in-process node as part of your application, but now you can use them in a running daemon as well.
+
+See the new [custom-ipld-formats](https://github.com/ipfs/js-ipfs/tree/master/examples/custom-ipld-formats) example in the `js-IPFS` repo for more information on how to set this up.
+
+## 🍏 ipfs-core
```

Comment:
"ipfs-core - new slimmer IPFS for library lovers" or something descriptive to get readers reading

Reply:
Done